asm_utils/src/data_storage.rs

```diff
@@ -0,0 +1,80 @@
+//! Utilites for generating initialization code that stores data objects in memory.
+
+use std::collections::BTreeMap;
+
+use crate::{data_parser::DataValue, utils::next_multiple_of_four};
+
+pub enum SingleDataValue<'a> {
+    Value(u32),
+    LabelReference(&'a String),
+    Offset(&'a String, &'a String),
+}
+
+pub fn store_data_objects<'a>(
+    objects: impl IntoIterator<Item = &'a (String, Vec<DataValue>)> + Copy,
+    mut memory_start: u32,
+    code_gen: &mut dyn FnMut(u32, SingleDataValue) -> Vec<String>,
+) -> (Vec<String>, BTreeMap<String, u32>) {
+    memory_start = ((memory_start + 7) / 8) * 8;
+    let mut current_pos = memory_start;
```

Comment:
```suggestion
    let mut current_pos = ((memory_start + 7) / 8) * 8;
```

then `memory_start` doesn't need to be mutable?